### DISCUSSION TITLE
Multiple Ashley IteratingSystems based on different families behave strangely

### DISCUSSION THREAD

**Discussion by Naliwe** [libktx/ktx#144] [Q&A] [answered]
Hello!

I've been trying to use 2 different IteratingSystems based on two different families of components and as soon as I have the second system **just written in code, not even instantiated or anything**, my first system doesn't get called at all:

```kotlin
val graphicFamily = allOf(Transform::class).oneOf(Text::class, Sprite::class)
val eventFamily = allOf(Transform::class, EventHandler::class)

class GraphicSystem : IteratingSystem(graphicFamily.get()) {
...
}

class EventSystem : IteratingSystem(eventFamily.get()) {
...
}
```

If I just comment out my EventSystem, the Graphic system picks up entities again. Is this by design ?

Thanks

**Comment by Naliwe** [comment, marked as answer]
Hm, nevermind, didn't think of calling `get` during family creation. Works great now. Maybe good to know that breaks everything though :3 

Sorry for the almost useless issue x)

**Comment by czyzby** [comment]
It's OK. :) If you have any further issues with Ashley itself, it might be easier to get help on [libgdx/ashley repository](https://github.com/libgdx/ashley), though - `ktx-ashley` it's only a thin utility wrapper over the library.
